### ISSUE TITLE
Migration from 1.2.5 to 1.2.6 fails

### ISSUE THREAD

**Issue by bskertchly** [snipe/snipe-it#586]
When I try to run artisan migrate, I get the following:
    [Illuminate\Database\QueryException]
    SQLSTATE[42S21]: Column already exists: 1060 Duplicate column name 'deleted
    _at' (SQL: alter table `users` add `deleted_at` timestamp null, add `websit
    e` varchar(255) null, add `country` varchar(255) null, add `gravatar` varch
    ar(255) null)

It's odd, because the the rest of the columns in the ALTER TABLE are there as well. It looks like it thinks my DB is at a much older version than what it really is.

Running on Ubuntu 14.04, installed using the snipeit-ansible scripts. 1.2.5 was my first install.

Any help would be greatly appreciated.

**Comment by bskertchly** [comment]
I'm not familiar with Laravel, but after some googling, I found that it determines what migrations to run based on the contents of the migrations table in the database. For some reason, this table is empty on my install.

**Comment by snipe** [comment]
Did you hit an error during the 1.2.5 original install that might have made it quit before the entire setup was done?

**Comment by bskertchly** [comment]
No errors on the original setup. The application was working just fine all last week. I'm trying to manually recreate the migrations table, but I'm not sure what should go in the migration field. Is it something like 2013_03_23_193214_update_users_table?

**Comment by bskertchly** [comment]
OK, so I inserted that first row into the DB (i.e. INSERT migrations(migration,batch) VALUES(2013_03_23_193214_update_users_table,1)), and I get an error related to the next migration, meaning it's working. I'll just go ahead and fix up the table. I'm still confused as to how the DB was created successfully, yet the migrations table was empty. From what I've read in the past hour, the DB is created through the migrations. Maybe I'm missing something. Anyway, thank you so much for your help. Your app is fantastic.

**Comment by uberbrady** [comment]
It's weird, I think we've seen this once before - where there's a db with schema but no migrations.

If there is any way you can think of to figure out how it may have gotten like that, it might help out the next person who did it that way.

And excellent Googlery btw. Very nicely done.

**Comment by bskertchly** [comment]
Thanks, Brady.

I can't think of anything that could have gotten it into that state.

Here's how I installed it:
-Followed instructions on [snipeit-ansible](https://github.com/GR360RY/snipeit-ansible) 
-Seeded the database (php artisan db:seed)

When I upgraded:
-Ran the Ansible task from snipeit-ansible to get the new version.
-Ran the composer upgrade, dump-autoload and artisan migrate scripts. Migrate failed.

To fix it:
-Went to /opt/snipe-it/app/database/migrations and did an ls>migrations.txt
-Worked my way up from the last migration to find the last migration that was applied.
-Constructed my insert statements and ran them against the database.

**Comment by uberbrady** [comment]
Bug reports this detailed make me very, very happy. Spectacular work. Thank you so much!

I'm wondering if it's some issue with the Ansible task(s) that somehow is blowing out the migration tables?

**Comment by bskertchly** [comment]
Thank you for the wonderful app! I'm glad I if I can contribute in some small way. Also, expect a donation soon once I figure out the financial stuff.

So, last night, I spun up a new Ubuntu VM and kept resetting it back to stock to try to reproduce the issue.

What I tried:
First run: Tried to reproduce the exact sequence of events described in my previous post. To simulate the upgrade, I modified the Ansible task to pick up the 1.2.5 release from GitHub, then modified it again to pick up the 1.2.6 release. Bug was not reproduced, everything worked as expected.

I did notice that the Ansible task doesn't reference a version, so it just picks up the latest source.

Second - Fifth runs:
Picked up each version on the develop branch between the 1.2.5 commit to master and the 1.2.6 master. Each installation went without a hitch. (I have to admit I gave up after the fourth version, because I didn't see this path going anywhere).

Sixth run:
Picked up a random 1.2.5 develop version, then upgraded to the latest 1.2.6 develop version (which most closely resembles what actually happened). Again, clean install and upgrade.

I'm sorry I can't reproduce the issue. I'm stumped. I think I must have either skipped something on the original install, or something happened between the original install and the upgrade which I'm not able to reproduce.

I'm going to suggest to the guys at snipeit-ansible to add a version=master condition to their task so that you always get the stable release.

I hope I haven't wasted too much of your time on this. Thanks again.
